bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java

```diff
@@ -225,25 +230,25 @@ private List<Path> listFiles(Path path, boolean includeSubDirectory) {
     public void processWatchEvent(WatchService.Kind kind, Path path) {
         Path fullPath = watchPath.resolve(path);
         File file = fullPath.toFile();
-        if (!file.isHidden()) {
-            if (kind == DELETE) {
-                if (file.isDirectory()) {
-                    if (watchSubDirectories) {
-                        synchronized (this) {
-                            String prefix = fullPath.getParent().toString();
-                            Set<String> toRemove = scriptMap.keySet().stream().filter(ref -> ref.startsWith(prefix))
-                                    .collect(Collectors.toSet());
-                            toRemove.forEach(this::removeFile);
-                        }
+
+        if (kind == DELETE) {
+            if (file.isDirectory()) {
```

Comment:
I wonder if this check is necessary. Did you check if we receive events for each file in the deleted directory? In that case we could ignore directory changes at all.

Reply:
You are absolutely right! I did some tests and deleting a directory with scripts inside it would correctly issue a DELETE event for each file inside that directory, but not the directory itself. This is because it's filtered by WatchServiceImpl so processWatchEvent would never receive events about directories anyway, so yes this piece of code will never be reached.

I have removed it and also adjusted the corresponding test.